src/transformers/models/idefics/configuration_idefics.py

```diff
@@ -163,22 +156,40 @@ def __init__(
         self.vision_intermediate_size = vision_intermediate_size
         self.vision_patch_size = vision_patch_size
 
-        if vision_config is None:
-            self.vision_config = {}
-            self.vision_config["hidden_size"] = vision_embed_dim
-            self.vision_config["image_size"] = vision_image_size
-            self.vision_config["num_attention_heads"] = num_attention_heads
-            self.vision_config["num_hidden_layers"] = num_hidden_layers
-            self.vision_config["intermediate_size"] = vision_intermediate_size
-            self.vision_config["patch_size"] = vision_patch_size
-        elif not isinstance(vision_config, dict):
-            raise ValueError("vision_config must be a dict")
-        else:
-            self.vision_config = vision_config
+        self.vision_config_dict = {}
+        self.vision_config_dict["hidden_size"] = vision_embed_dim
+        self.vision_config_dict["image_size"] = vision_image_size
+        self.vision_config_dict["num_attention_heads"] = vision_num_attention_heads
+        self.vision_config_dict["num_hidden_layers"] = vision_num_hidden_layers
+        self.vision_config_dict["intermediate_size"] = vision_intermediate_size
+        self.vision_config_dict["patch_size"] = vision_patch_size
 
         # Resampler params
         self.use_resampler = use_resampler
         self.resampler_n_latents = resampler_n_latents
         self.resampler_depth = resampler_depth
         self.resampler_n_heads = resampler_n_heads
         self.resampler_head_dim = resampler_head_dim
+        # IMPORTANT: Do not do any __init__ args-based checks in the constructor, since
+        # PretrainedConfig.from_dict first instantiates the class with the config dict and only then
+        # updates the config object with `kwargs` from from_pretrained, so during the instantiation
+        # of this object many attributes have default values and haven't yet been overridden.
+        # Do any required checks inside `from_pretrained` once the superclass' `from_pretrained` was run.
+        super().__init__(
+            pad_token_id=pad_token_id,
+            bos_token_id=bos_token_id,
+            eos_token_id=eos_token_id,
+            tie_word_embeddings=tie_word_embeddings,
+            **kwargs,
+        )
+
```

Comment:
change order, add some breathing space

```suggestion

        super().__init__(
            pad_token_id=pad_token_id,
            bos_token_id=bos_token_id,
            eos_token_id=eos_token_id,
            tie_word_embeddings=tie_word_embeddings,
            **kwargs,
        )

        # IMPORTANT: Do not do any __init__ args-based checks in the constructor, since
        # PretrainedConfig.from_dict first instantiates the class with the config dict and only then
        # updates the config object with `kwargs` from from_pretrained, so during the instantiation
        # of this object many attributes have default values and haven't yet been overridden.
        # Do any required checks inside `from_pretrained` once the superclass' `from_pretrained` was run.
```